src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxTests.cs

```diff
@@ -276,5 +279,54 @@ public void IsAttributeTargetSpecifier()
                 }
             }
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72300")]
+        public void TestAllKindsReturnedFromGetKindsMethodsExist()
+        {
+            foreach (var method in typeof(SyntaxFacts).GetMethods(BindingFlags.Public | BindingFlags.Static))
+            {
+                if (method.ReturnType == typeof(IEnumerable<SyntaxKind>) && method.GetParameters() is [])
+                {
+                    foreach (var kind in (IEnumerable<SyntaxKind>)method.Invoke(null, null))
+                    {
+                        Assert.True(Enum.IsDefined(typeof(SyntaxKind), kind), $"Nonexistent kind '{kind}' returned from method '{method.Name}'");
+                    }
+                }
+            }
+        }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/72300")]
+        [InlineData(nameof(SyntaxFacts.GetContextualKeywordKinds), SyntaxKind.YieldKeyword, SyntaxKind.FileKeyword)]
+        [InlineData(nameof(SyntaxFacts.GetPreprocessorKeywordKinds), SyntaxKind.ElifKeyword, SyntaxKind.RestoreKeyword)] // This case doesn't cover all possible preprocessor kinds
+        [InlineData(nameof(SyntaxFacts.GetPunctuationKinds), SyntaxKind.TildeToken, SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken)]
+        [InlineData(nameof(SyntaxFacts.GetReservedKeywordKinds), SyntaxKind.BoolKeyword, SyntaxKind.ImplicitKeyword)]
+        public void TestRangeBasedGetKindsMethodsReturnExpectedResults(string methodName, SyntaxKind firstExpectedKind, SyntaxKind lastExpectedKind)
+        {
+            var method = typeof(SyntaxFacts).GetMethod(methodName, BindingFlags.Public | BindingFlags.Static);
+
+            Assert.NotNull(method);
+            Assert.Equal(0, method.GetParameters().Length);
+            Assert.Equal(typeof(IEnumerable<SyntaxKind>), method.ReturnType);
+
+            var kindsFromMethod = ((IEnumerable<SyntaxKind>)method.Invoke(null, null)).ToArray();
+
+            // Check that the value right before the expected lower bound isn't contained in the enumerable returned from the target method
+            Assert.DoesNotContain((SyntaxKind)((int)firstExpectedKind - 1), kindsFromMethod);
```

Comment:
>Assert.DoesNotContain((SyntaxKind)((int)firstExpectedKind - 1), kindsFromMethod); [](http://example.com/codeflow?start=12&length=81)

It is not clear why we are checking only against two values outside of the range. What prevents us from iterating through the result and asserting all values are within the expected range?

Reply:
> It is not clear why we are checking only against two values outside of the range. What prevents us from iterating through the result and asserting all values are within the expected range?

I am not sure what has happened with this suggestion. It looks like instead of strengthening the check. I.e. making sure the returned kinds all are withing the expected range, the following checks were completely removed:
```
            // Check that the value right before the expected lower bound isn't contained in the enumerable returned from the target method
            Assert.DoesNotContain((SyntaxKind)((int)firstExpectedKind - 1), kindsFromMethod);
             // Check that the value right after the expected upper bound isn't contained in the enumerable returned from the target method
            Assert.DoesNotContain((SyntaxKind)((int)lastExpectedKind + 1), kindsFromMethod);
```

Could you, please, elaborate?

Reply:
I initialy understood this review as "we can replace current logic with simple in-range check". After doing that you said that the tests are no longer representative since they won't fail if the method fails to return all expected kinds. At the point of partially reverting my changes I just forgot that this was a thing.

Reply:
Done by removing expected members from the set and at the end asserting, that the set is empty. It does the same thing as "in range" check since if something isn't in range it is gonna be leftover in the set and thus the assertion would fail